<?xml version="1.0" encoding="UTF-8"?>
<!--
    $Author: DJ  
    $Date: 2013-4-1 上午9:37:24  
    $Revision: 1.0  
    $Purpose: 
-->
<bm:model xmlns:e="leaf.service.exception" xmlns:bm="http://www.leaf-framework.org/schema/bm" needAccessControl="false">
    <bm:operations>
        <bm:operation name="query">
            <bm:query-sql><![CDATA[
                SELECT
                    *
                FROM
                    (SELECT
                        h.score_target_id,
                        fnd.*
                    FROM
                        fnd_sc_score_result_dtl fnd,
                        fnd_score_template_line h
                    WHERE
                        fnd.score_template_line_id = h.score_template_line_id
                    ) #WHERE_CLAUSE#
            ]]></bm:query-sql>
        </bm:operation>
        <bm:operation name="update">
            <bm:update-sql><![CDATA[
            	UPDATE
            	    fnd_sc_score_result_dtl h
            	SET
            	    h.target_value = ${@target_value}
            	WHERE
            	    h.result_id              = ${@result_id} AND
            	    h.score_template_line_id =
            	    (SELECT
            	        l.score_template_line_id
            	    FROM
            	        fnd_score_template_line l
            	    WHERE
            	        l.score_target_id   = ${@score_target_id} AND
            	        l.score_template_id = ${@score_template_id}
            	    )
        	]]></bm:update-sql>
        </bm:operation>
    </bm:operations>
    <bm:query-fields>
        <bm:query-field name="score_target_id" queryExpression="score_target_id=${@score_target_id}"/>
        <bm:query-field name="result_id" queryExpression="result_id=${@result_id}"/>
    </bm:query-fields>
    <e:exception-descriptor-config>
        <e:exception-descriptor exception="java.sql.SQLException" handleClass="leaf.database.SQLExceptionDescriptor">
            <e:error-message code="1" message="FND712.ERROR_MESSAGE.SCORE_TARGET_CODE_REPEATED"/>
        </e:exception-descriptor>
    </e:exception-descriptor-config>
</bm:model>
